tests/instrument/expr_match.ml.reference

```diff
@@ -99,3 +99,174 @@ let f x =
+type v = {
+  a: t;
+  b: t;}
+let f x =
+  ___bisect_mark___expr_match 78;
+  (match x with
+   | { a = (Foo |Bar ); b = (Foo |Bar ) } as ___bisect_matched_value___ ->
+       ((((match ___bisect_matched_value___ with
+           | { a = Foo ; b = Foo  } ->
+               (___bisect_mark___expr_match 75;
+                ___bisect_mark___expr_match 74;
+                ())
+           | { a = Foo ; b = Bar  } ->
+               (___bisect_mark___expr_match 76;
+                ___bisect_mark___expr_match 74;
+                ())
+           | { a = Bar ; b = Foo  } ->
+               (___bisect_mark___expr_match 75;
+                ___bisect_mark___expr_match 77;
+                ())
+           | { a = Bar ; b = Bar  } ->
+               (___bisect_mark___expr_match 76;
+                ___bisect_mark___expr_match 77;
+                ())
+           | _ -> ()))[@ocaml.warning "-8-11"]);
+        print_endline "foo"))
+let f x =
+  ___bisect_mark___expr_match 86;
+  (match x with
+   | [||] -> (___bisect_mark___expr_match 79; print_endline "foo")
+   | [|(Foo |Bar );(Foo |Bar );_|] as ___bisect_matched_value___ ->
+       ((((match ___bisect_matched_value___ with
+           | [|Foo ;Foo ;_|] ->
+               (___bisect_mark___expr_match 82;
+                ___bisect_mark___expr_match 81;
+                ___bisect_mark___expr_match 80;
+                ())
+           | [|Foo ;Bar ;_|] ->
+               (___bisect_mark___expr_match 83;
+                ___bisect_mark___expr_match 81;
+                ___bisect_mark___expr_match 80;
+                ())
+           | [|Bar ;Foo ;_|] ->
+               (___bisect_mark___expr_match 82;
+                ___bisect_mark___expr_match 84;
+                ___bisect_mark___expr_match 80;
+                ())
+           | [|Bar ;Bar ;_|] ->
+               (___bisect_mark___expr_match 83;
+                ___bisect_mark___expr_match 84;
+                ___bisect_mark___expr_match 80;
+                ())
+           | _ -> ()))[@ocaml.warning "-8-11"]);
+        print_endline "bar")
+   | _ -> (___bisect_mark___expr_match 85; print_newline ()))
+let f x =
+  ___bisect_mark___expr_match 89;
+  (match x with
+   | (lazy (Foo |Bar )) as ___bisect_matched_value___ ->
+       ((((match ___bisect_matched_value___ with
+           | (lazy Foo ) -> (___bisect_mark___expr_match 87; ())
+           | (lazy Bar ) -> (___bisect_mark___expr_match 88; ())
+           | _ -> ()))[@ocaml.warning "-8-11"]);
+        print_endline "foo"))
+exception Exn of t
+let f x =
+  ___bisect_mark___expr_match 93;
+  (match x with
+   | exception (Exn (Foo |Bar ) as ___bisect_matched_value___) ->
+       ((((match ___bisect_matched_value___ with
+           | Exn (Foo ) -> (___bisect_mark___expr_match 90; ())
+           | Exn (Bar ) -> (___bisect_mark___expr_match 91; ())
+           | _ -> ()))[@ocaml.warning "-8-11"]);
+        print_endline "foo")
+   | _ -> (___bisect_mark___expr_match 92; print_endline "bar"))
```

Comment:
You're right this is not needed. I saw this as a case requiring two levels of recursion, while your previous examples (as far as I understood) were all one level. But because of the coverage of our coverage tool, you had all the corner cases tested!